packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFScrollViewDelegateHostApi.m

```diff
@@ -0,0 +1,90 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FWFScrollViewDelegateHostApi.h"
+#import "FWFWebViewHostApi.h"
+
+@interface FWFScrollViewDelegateFlutterApiImpl ()
+// BinaryMessenger must be weak to prevent a circular reference with the host API it
+// references.
+@property(nonatomic, weak) id<FlutterBinaryMessenger> binaryMessenger;
+// InstanceManager must be weak to prevent a circular reference with the object it stores.
+@property(nonatomic, weak) FWFInstanceManager *instanceManager;
+@end
+
+@implementation FWFScrollViewDelegateFlutterApiImpl
+
+- (instancetype)initWithBinaryMessenger:(id<FlutterBinaryMessenger>)binaryMessenger
+                        instanceManager:(FWFInstanceManager *)instanceManager {
+  self = [self initWithBinaryMessenger:binaryMessenger];
+  if (self) {
+    _binaryMessenger = binaryMessenger;
+    _instanceManager = instanceManager;
+  }
+  return self;
+}
+- (long)identifierForDelegate:(FWFScrollViewDelegate *)instance {
+  return [self.instanceManager identifierWithStrongReferenceForInstance:instance];
+}
+
+- (void)scrollViewDidScrollForDelegate:(FWFScrollViewDelegate *)instance
+                          uiScrollView:(UIScrollView *)scrollView
+                            completion:(void (^)(FlutterError *_Nullable))completion {
+  [self scrollViewDidScrollWithIdentifier:[self identifierForDelegate:instance]
+                   uiScrollViewIdentifier:[self.instanceManager
+                                              identifierWithStrongReferenceForInstance:scrollView]
+                                        x:scrollView.contentOffset.x
+                                        y:scrollView.contentOffset.y
+                               completion:completion];
+}
+@end
+
+@implementation FWFScrollViewDelegate
+
+- (instancetype)initWithBinaryMessenger:(id<FlutterBinaryMessenger>)binaryMessenger
+                        instanceManager:(FWFInstanceManager *)instanceManager {
+  self = [super initWithBinaryMessenger:binaryMessenger instanceManager:instanceManager];
+  if (self) {
+    _scrollViewDelegateAPI =
+        [[FWFScrollViewDelegateFlutterApiImpl alloc] initWithBinaryMessenger:binaryMessenger
+                                                             instanceManager:instanceManager];
+  }
+  return self;
+}
+
+- (void)scrollViewDidScroll:(UIScrollView *)scrollView {
+  [self.scrollViewDelegateAPI scrollViewDidScrollForDelegate:self
+                                                uiScrollView:scrollView
+                                                  completion:^(FlutterError *error) {
+                                                    NSAssert(!error, @"%@", error);
```

Comment:
should we pipe the error to flutter side?

Reply:
As I understand, the `completion` callback here is for the native side to handle the error pipe from the flutter side, so I don't think it is logical to passing it back to the flutter side. Besides, every callback that was implemented before me is doing the same, I just followed the convention.